test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -567,6 +569,82 @@ TEST(ParticipantTests, CreateDomainParticipantWithProfile)
     ASSERT_TRUE(DomainParticipantFactory::get_instance()->delete_participant(participant) == ReturnCode_t::RETCODE_OK);
 }
 
+TEST(ParticipantTests, CreateDomainParticipantWithDefaultProfile)
+{
+    uint32_t domain_id = 123u;          // This is the domain ID set in the default profile above
+
+    // set XML profile as environment variable: "export FASTDDS_DEFAULT_PROFILES_FILE=test_xml_profile.xml"
+    eprosima::testing::set_environment_variable("FASTDDS_DEFAULT_PROFILES_FILE", "test_xml_profile.xml");
+
+    //participant using the given profile
+    DomainParticipant* default_env_participant =
+            DomainParticipantFactory::get_instance()->create_participant_with_default_profile();
+
+    // unset XML profile environment variable
+    eprosima::testing::clear_environment_variable("FASTDDS_DEFAULT_PROFILES_FILE");
+
+    ASSERT_NE(default_env_participant, nullptr);
+    ASSERT_EQ(default_env_participant->get_domain_id(), domain_id);
+    ASSERT_EQ(default_env_participant->get_listener(), nullptr);
+    ASSERT_TRUE(DomainParticipantFactory::get_instance()->delete_participant(
+                default_env_participant) == ReturnCode_t::RETCODE_OK);
+}
+
+TEST(ParticipantTests, CreateDomainParticipantWithDefaultProfileListener)
+{
+    uint32_t domain_id = 123u;          // This is the domain ID set in the default profile above
+
+    // set XML profile as environment variable: "export FASTDDS_DEFAULT_PROFILES_FILE=test_xml_profile.xml"
+    eprosima::testing::set_environment_variable("FASTDDS_DEFAULT_PROFILES_FILE", "test_xml_profile.xml");
+
+    DomainParticipantListener* listener = new DomainParticipantListener();
+
+    //participant using the given profile
+    DomainParticipant* default_env_participant =
+            DomainParticipantFactory::get_instance()->create_participant_with_default_profile(listener,
+                    StatusMask::none());
+
+    // unset XML profile environment variable
+    eprosima::testing::clear_environment_variable("FASTDDS_DEFAULT_PROFILES_FILE");
+
+    ASSERT_NE(default_env_participant, nullptr);
+    ASSERT_EQ(default_env_participant->get_domain_id(), domain_id);
+    ASSERT_EQ(default_env_participant->get_listener(), listener);
```

Comment:
The `listener` is not being deleted, which causes a memory leak in the test and thus ASan job fails. I suggest instantiating the listener on stack instead.
```suggestion
    DomainParticipantListener listener;

    //participant using the given profile
    DomainParticipant* default_env_participant =
            DomainParticipantFactory::get_instance()->create_participant_with_default_profile(&listener,
                    StatusMask::none());

    // unset XML profile environment variable
    eprosima::testing::clear_environment_variable("FASTDDS_DEFAULT_PROFILES_FILE");

    ASSERT_NE(default_env_participant, nullptr);
    ASSERT_EQ(default_env_participant->get_domain_id(), domain_id);
    ASSERT_EQ(default_env_participant->get_listener(), &listener);
```